packages/design-system-twrnc-preset/src/ThemeProvider.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useMemo } from 'react';
+import { create } from 'twrnc';
+
+import { generateTailwindConfig } from './tailwind.config';
+import type { Theme } from './Theme.types';
+import type { ThemeContextProps } from './ThemeContext';
+import { ThemeContext } from './ThemeContext';
+
+/**
+ * Theme provider component that wraps child components with theme context
+ *
+ * @param options - Component props
+ * @param options.children - Child components to render
+ * @param options.theme - Theme to apply (light or dark)
+ * @returns React component that provides theme context to children
+ */
+export const ThemeProvider = ({
+  children,
+  theme,
+}: {
+  children: React.ReactNode;
+  theme: Theme;
+}) => {
+  const contextValue: ThemeContextProps = useMemo(() => {
+    const tailwindConfig = generateTailwindConfig(theme);
+    const tw = create(tailwindConfig);
+    return {
+      tw,
+      theme,
+    };
+  }, [theme]);
+
+  return (
+    <ThemeContext.Provider value={contextValue}>
+      {children}
+    </ThemeContext.Provider>
+  );
+};
```

Comment:
Dramatically simplified the `ThemeProvider` component from a complex 70-line stateful component that managed `ColorSet`, `ColorScheme`, auto-detection with `useColorScheme()`, and multiple state setters (`setColorSet`, `setTheme`) into a clean component that simply takes a `theme` prop, generates the appropriate Tailwind config using `useMemo`, creates the `twrnc` instance, and provides it through context - removing all the internal state management and theme detection logic in favor of making it a pure, controlled component that consumers manage externally.